### PR TITLE
Fix: e2e tests to check _canCycleThroughPagination (fixes #312)

### LINKED ISSUE #312: e2e test fails if _canCycleThroughPagination is set to true
### Subject of the issue/enhancement/features
Describe your issue here.

### Your environment
* Latest FW

### Steps to reproduce
Create a course with a hotgraphic, set _canCycleThroughPagination to true and run npm test e2e

### PR DESCRIPTION
**Pull request by joe-allen-89**
[//]: # (Please title your PR according to eslint commit conventions)
[//]: # (See https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-eslint#eslint-convention for details)

[//]: # (Link the PR to the original issue)
#312 

[//]: # (Delete Fix, Update, New and/or Breaking sections as appropriate)
### Fix
* e2e tests to check _canCycleThroughPagination (fixes #312)

### CODE REVIEW

**Review by lemmyadams — APPROVED** [review]

**Review by oliverfoster — APPROVED** [review]

File: test/e2e/hotGraphic.cy.js [hunk 1]
```diff
@@ -16,18 +17,29 @@ describe('Hot Graphic', function () {
...
       cy.get('.hotgraphic-popup__controls.next').click();
-    })
-    cy.get('.hotgraphic-popup__controls.next.is-disabled').should('exist');
-    items.forEach(item =>  {
+    });
+
+    if (!canCycle) {
+      cy.get('.hotgraphic-popup__controls.next.is-disabled').should('exist');
+    } else {
+      cy.get('.hotgraphic-popup__controls.next').should('exist');
```
Comment (cahirodoherty-learningpool) [inline test/e2e/hotGraphic.cy.js:29]:
```suggestion
      cy.get('.hotgraphic-popup__controls.next').should('exist');
      cy.get('.hotgraphic-popup__controls.next.is-disabled').should('not.exist');
```

File: test/e2e/hotGraphic.cy.js [hunk 2]
```diff
@@ -16,18 +17,29 @@ describe('Hot Graphic', function () {
...
       cy.testContainsOrNotExists('.hotgraphic-popup__item-body-inner', item.body);
       cy.get('.hotgraphic-popup__controls.back').click();
-    })
-    cy.get('.hotgraphic-popup__controls.back.is-disabled').should('exist');
+    });
+
+    if (!canCycle) {
+      cy.get('.hotgraphic-popup__controls.back.is-disabled').should('exist');
+    } else {
+      cy.get('.hotgraphic-popup__controls.back').should('exist');
```
Comment (cahirodoherty-learningpool) [inline test/e2e/hotGraphic.cy.js:41]:
```suggestion
      cy.get('.hotgraphic-popup__controls.back').should('exist');
      cy.get('.hotgraphic-popup__controls.back.is-disabled').should('not.exist');
```

File: test/e2e/hotGraphic.cy.js (hunk 1 above)
Comment (joe-allen-89) [reply test/e2e/hotGraphic.cy.js:29]: Do we need to check if .next.is-disabled does not exist because if .next exists then .next.is-disabled can't?

Comment (cahirodoherty-learningpool) [reply test/e2e/hotGraphic.cy.js:29]: I might be wrong here, but I don't think the existence of a `.hotgraphic-popup__controls.next` excludes the possibility of that element also having the class `is-disabled` applied to it
So by doing 2 Cypress checks we are confirming the expected element exists, but confirming that it does not have the `is-disabled` class

Comment (lemmyadams) [reply test/e2e/hotGraphic.cy.js:29]: ^ I think we could probably do it in one command or something to check is the `.hotgraphic-popup__controls.next` exists and doesn't have the `is-disabled` class as well. Not sure exactly on the command there though

Comment (cahirodoherty-learningpool) [reply test/e2e/hotGraphic.cy.js:29]:
```suggestion
      cy.get('.hotgraphic-popup__controls.next')
        .should('exist')
        .and('not.have.class', 'is-disabled')
```

File: test/e2e/hotGraphic.cy.js (hunk 2 above)
Comment (cahirodoherty-learningpool) [reply test/e2e/hotGraphic.cy.js:41]:
```suggestion
      cy.get('.hotgraphic-popup__controls.back')
        .should('exist')
        .and('not.have.class', 'is-disabled')
```

**Comment by cahirodoherty-learningpool** [comment]
I have updated my code suggestions to be a bit more concise following Lemmy's idea

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**